addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1142,9 +1142,21 @@ export class Wysiwyg extends Component {
             // o_modified_image_to_save by _saveB64Image to request the saving
             // of the pre-converted webp resizes and all the equivalent jpgs.
             const b64Proms = [...editableEl.querySelectorAll('.o_b64_image_to_save')].map(async el => {
+                if (!el.closest(".o_dirty").isEqualNode(editableEl)) {
+                    // Do nothing as there is an editableEl closer to the image
+                    // that will perform the `_saveB64Image()` with the correct
+                    // parameters.
+                    return;
+                }
                 await this._saveB64Image(el, resModel, resId);
             });
             const modifiedProms = [...editableEl.querySelectorAll('.o_modified_image_to_save')].map(async el => {
+                if (!el.closest(".o_dirty").isEqualNode(editableEl)) {
+                    // Do nothing as there is an editableEl closer to the image
+                    // that will perform the `_saveModifiedImage()` with the
+                    // correct parameters.
+                    return;
+                }
                 await this._saveModifiedImage(el, resModel, resId);
             });
             return Promise.all([...b64Proms, ...modifiedProms]);
```

Comment:
Note : This will be done in this [PR] as it should be a fix in stable as well

[PR]: https://github.com/odoo/odoo/pull/130472